src/units/abilities.cpp

```diff
@@ -751,6 +751,10 @@ namespace {
  */
 bool attack_type::has_special(const std::string& special, bool simple_check, bool special_id, bool special_tags) const
 {
+	if(!simple_check && num_recursion >=3){
```

Comment:
Why 3? It may be correct, but I'm wondering why that particular value was chosen.

Reply:
0 is the value when has_special directly called like weapon. 1 value when called with a filter_weapon inside a weapon special, the limit for force exit of loop should be of 2 but i prefer for now let to 3 for be sure what it is a infinite recursion.

I not usre myself is i should use 2 or 3, the diffculty for encoded the  code it done foreget myself to test other value of 3.

Reply:
changed for 2